core/src/main/java/com/graphhopper/util/PathSimplification.java

```diff
@@ -98,6 +98,16 @@ public PointList simplify() {
                     if (removed > 0) {
                         for (int i = 0; i < toSimplify.size(); i++) {
                             reduceLength(toSimplify.get(i), offset[i], startIntervals[i], endIntervals[i] - removed);
+                            /*
```

Comment:
> Ah, I think I have now fully understood the complexity, so when there are n PathDetails we do n simplifications

No, we only have one PointList that we simplify. We only update the PahtDetails and Instructions. That's why I would like to remove the Points from the Instructions :).

> roughly n²/2 updates to the PathDetail object

No, as I said, we don't need to propagate the changes. Everything works with this code commented out. At first I thought it would be necessary though.

> The only alternative would be to sort the PathDetails by first index, create a dependency graph and then being to only update the PathDetails that are necessary

Instead of doing the simplififcation per PathDetails I tried to do it one run, which makes the process a bit more complicated but saves us a lot of computation (I think).